docs/src/proposals/rip-curl.md

```diff
@@ -0,0 +1,56 @@
+# RiP Curl: low-latency, transaction-oriented RPC
+
+## Problem
+
+Solana's initial RPC implementation was created for the purpose of allowing
+users to confirm transactions that had just recently been sent to the cluster.
+It was designed with memory usage in mind such that any validator should be
+able to support the API without concern of DoS attacks.
+
+Later down the line, it became desirable to use that same API to support the
+Solana explorer. The original design only supported minutes of history, so we
+changed it to instead store transaction statuses in a local RocksDB instance
+and offer days of history. We then extended that to 6 months via BigTable.
+
+With each modification, the API became more suitable for explorers and less
```

Comment:
I think this assumes that explorers only serve static content. This is likely because the current explorer.solana.com is too static. It should be dynamic and subscribe to event streams just like a client would for tx processing.

I think the problem is just that there hasn't been enough focus on stream based APIs. There's a place for historical query APIs too, but I expect most apps (explorer included) should be architected to consume live event streams

Reply:
That's a great point.  I should say "the API became more suitable for serving static content".